mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -445,6 +445,13 @@ class WidgetAdapter(
             boundWidget = widget
 
             val stepCount = (widget.maxValue - widget.minValue) / widget.step
+
+            if (stepCount <= DISCRETE_SLIDER_STATES_THRESHOLD) {
+                seekBar.context.setTheme(R.style.Widget_AppCompat_SeekBar_Discrete)
+            } else {
+                seekBar.context.setTheme(R.style.Widget_AppCompat_SeekBar)
```

Comment:
Changing the theme for the whole activity (that's what the view context is) isn't a viable solution, especially given a widget style (a collection of widget specific attributes) isn't the same thing as a theme (a collection of activity specific attributes).
Are you sure this even works? I fail to see how it could...

Reply:
Yes, you're right, changing activity theme is a bad option. It's possible to apply a style as a theme and it works but I agree with you that we mustn't do that. 

The ideal approach IMO would be constructing the seek bar programmatically passing the correspondent style in the constructor depending on the number of states. However, I didn't implemented that option because inside `SliderViewHolder` the seek bar is used before knowing the number of states it's going to have (there is code to set listeners on the `init` method of the view holder and it's also used on `handleRowClick`).

I could try to implement the programatical approach, but I would like to know your opinion on how should this be implemented before going for it.

Reply:
If reconfiguring the SeekBar between discrete and continuous at runtime doesn't work (maybe checking the Discrete style would give a clue what needs to be done there?), I'd likely go with two different view types and two layouts and distinguishing between them via SliderViewHolder constructor parameter.

Reply:
> If reconfiguring the SeekBar between discrete and continuous at runtime doesn't work

I tried to add the tick marks manually, but this doesn't work for Android < N: https://github.com/mueller-ma/openhab.android/commit/7606d828e1890e271e6170ad359034b6b8265f75

Reply:
What's the issue on < N?

Reply:
````
rg.openhab.habdroid.beta E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.openhab.habdroid.beta, PID: 8949
    java.lang.NoSuchMethodError: No virtual method setTickMark(Landroid/graphics/drawable/Drawable;)V in class Landroidx/appcompat/widget/AppCompatSeekBar; or its super classes (declaration of 'androidx.appcompat.widget.AppCompatSeekBar' appears in /data/app/org.openhab.habdroid.beta-2/base.apk)
        at org.openhab.habdroid.ui.WidgetAdapter$SliderViewHolder.bind(WidgetAdapter.kt:469)
        at org.openhab.habdroid.ui.WidgetAdapter.onBindViewHolder(WidgetAdapter.kt:187)
        at org.openhab.habdroid.ui.WidgetAdapter.onBindViewHolder(WidgetAdapter.kt:88)
        at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7059)
        at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7101)
````